src/ConfirmationDialog.js

```diff
@@ -51,6 +52,28 @@ const ConfirmationDialog = ({
     </>
   );
 
+  const dialogActions = buttonOrder.map(o => {
+    switch (o) {
+      case 'cancel':
+        if (!hideCancelButton) {
+          return (
+            <Button {...cancellationButtonProps} onClick={onCancel}>
+              {cancellationText}
+            </Button>
+          )
+        }
+      case 'confirm': 
```

Comment:
We should also raise an error for unknown value.

(Subjective: I usually prefer multiple ifs over switch because it's less nesting, but I'm fine with whichever)